cogs/moderation.py

```diff
@@ -327,14 +329,59 @@ async def execute(self, ctx):
         await super().execute(ctx)
 
 
+class EmergencyAlertBan(Action):
+    type = "emergency_alert_ban"
+    past_tense = "banned from emergency staff alerts"
+    emoji = "\N{BELL WITH CANCELLATION STROKE}"
+    color = discord.Color.magenta()
+
+    async def execute(self, ctx):
+        await ctx.bot.mongo.db.member.update_one(
+            {"_id": {"id": self.target.id, "guild_id": ctx.guild.id}},
+            {"$set": {"emergency_alert_banned_until": self.expires_at or True}},
```

Comment:
How this cog works is just that you set `emergency_alert_banned` to True, and then when the action expires it'll invoke the Unban which should set it back to false

Reply:
I understand that, but what this piece of code is doing is that it's using `emergency_alert_banned_until` for both temp and permanent bans. If it's temp (i.e. has the `expires_at` attr), it'll set that otherwise it'll set `True` (indefinite amount of time).